app/controllers/goals_controller.rb

```diff
@@ -6,8 +6,28 @@ def index
   end
 
   def create
-    Goal.create_goal(name: params[:name])
+    Goal.create_goal(goal_params)
     render json:{}, status: :created
   end
 
+  def update
+    current_goal = Goal.find(params[:id])
+    current_goal.update(goal_params)
+    render json:{}, status: :accepted
+  end
+
+  def destroy
+    current_goal = Goal.find(params[:id])
+    current_goal.destroy
+    render nothing: true, status: :no_content
+  end
+
+
+
+  private
+
+    def goal_params
```

Comment:
Does this work for JSON post requests? I thought it might only be for forms but may be wrong.